Extension/src/LanguageServer/client.ts

```diff
@@ -1980,123 +1999,129 @@ export class DefaultClient implements Client {
-            const tokenSource: vscode.CancellationTokenSource = new vscode.CancellationTokenSource();
-            console.log("provideCustomConfiguration");
+    private async provideCustomConfigurationAsync(docUri: vscode.Uri, requestFile: string | undefined, replaceExisting: boolean | undefined, onFinished: () => void, provider: CustomConfigurationProvider1) {
+        DefaultClient.isStarted.reset();
 
-            const providerName: string = provider.name;
+        const tokenSource: vscode.CancellationTokenSource = new vscode.CancellationTokenSource();
+        console.log("provideCustomConfiguration");
 
-            const params: QueryTranslationUnitSourceParams = {
-                uri: docUri.toString(),
-                ignoreExisting: !!replaceExisting,
-                workspaceFolderUri: this.RootUri?.toString()
-            };
-            const response: QueryTranslationUnitSourceResult = await this.languageClient.sendRequest(QueryTranslationUnitSourceRequest, params);
-            if (!response.candidates || response.candidates.length === 0) {
-                // If we didn't receive any candidates, no configuration is needed.
-                onFinished();
-                return;
-            }
+        const params: QueryTranslationUnitSourceParams = {
+            uri: docUri.toString(),
+            ignoreExisting: !!replaceExisting,
+            workspaceFolderUri: this.RootUri?.toString()
+        };
 
-            // Need to loop through candidates, to see if we can get a custom configuration from any of them.
-            // Wrap all lookups in a single task, so we can apply a timeout to the entire duration.
-            const provideConfigurationAsync: () => Thenable<SourceFileConfigurationItem[] | null | undefined> = async () => {
-                const uris: vscode.Uri[] = [];
-                for (let i: number = 0; i < response.candidates.length; ++i) {
-                    const candidate: string = response.candidates[i];
-                    const tuUri: vscode.Uri = vscode.Uri.parse(candidate);
-                    try {
-                        if (await provider.canProvideConfiguration(tuUri, tokenSource.token)) {
-                            uris.push(tuUri);
-                        }
-                    } catch (err) {
-                        console.warn("Caught exception from canProvideConfiguration");
-                    }
-                }
-                if (!uris.length) {
-                    return [];
-                }
-                let configs: util.Mutable<SourceFileConfigurationItem>[] = [];
+        const response: QueryTranslationUnitSourceResult = await this.languageClient.sendRequest(QueryTranslationUnitSourceRequest, params);
+        if (!response.candidates || response.candidates.length === 0) {
+        // If we didn't receive any candidates, no configuration is needed.
+            onFinished();
+            DefaultClient.isStarted.resolve();
```

Comment:
Future-proofing: it might be safer to wrap the whole function in try/finally to ensure the resolve always happens (in case any other early returns appear in here someday.